### ISSUE TITLE
Feature request: partial file reading

### ISSUE THREAD

**Issue by Mathis-Z** [nicholasbishop/ext4-view-rs#330]
Currently one can only read entire files. When working with large files however it would be nice to have an API to only read parts of a file. I think something like ```read(offset, length) -> Result<Vec<u8>, Ext4Error>``` would be nice.

**Comment by nicholasbishop** [comment]
I agree that would be useful. I'm thinking maybe introduce a new `File` type. If the `std` feature is enabled it can provide `Read` and `Seek` impls, and for non-std a simple `read` method with an interface like you suggested.
